pkg/api/types.go

```diff
@@ -1385,8 +1403,13 @@ func (a *AgentPoolProfile) GetKubernetesLabels(rg string) string {
 		buf.WriteString(fmt.Sprintf(",accelerator=%s", accelerator))
 	}
 	buf.WriteString(fmt.Sprintf(",kubernetes.azure.com/cluster=%s", rg))
-	for k, v := range a.CustomNodeLabels {
-		buf.WriteString(fmt.Sprintf(",%s=%s", k, v))
+	keys := []string{}
```

Comment:
These changes will fix the transient `TestAgentPoolProfileGetKubernetesLabels` unit test failures